typescript/lib/services/service_context.ts

```diff
@@ -0,0 +1,129 @@
+                    )];
+            }
+            // let absFilepathOnTestsuite = new URL(this.testVolumeMountpointOnTestsuiteContainer, relativeFilepath); //TODO need to confirm this VS path
+            // let absFilepathOnService = new URL(this.testVolumeMountpointOnServiceContainer, relativeFilepath);
+            let absFilepathOnTestsuite = path.Join(this.testVolumeMountpointOnTestsuiteContainer, relativeFilepath);
+            let absFilepathOnService = path.Join(this.testVolumeMountpointOnServiceContainer, relativeFilepath);
+            let genFilePath = new GeneratedFileFilepaths(); //TODO = this might be a problem in terms of memory (direct or reference), so check
+            genFilePath.absoluteFilepathOnTestsuiteContainer = absFilepathOnTestsuite;
+            genFilePath.absoluteFilepathOnServiceContainer = absFilepathOnService;
+            result.set(fileId, genFilePath); 
+            //}
+        }
+        return [result, null]; //TODO - Design decision, returning multiple values as an array
+    }
+
+    // Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+    public loadStaticFiles(usedStaticFilesSet: Set<StaticFileID>): [Map<StaticFileID, string>, Error] { // TODO (comment) - changed Map into Set; possible to return useful erorr messages in typescript
+        let serviceId = this.serviceId;
+        let staticFilesToCopyStringSet = new Map(); //TODO (comment) = map[string]bool{} ; kept this as a map sinc call to loadStaticFile has map property that requires a boolean
+        for (let staticFileId in usedStaticFilesSet) {
+            staticFilesToCopyStringSet[String(staticFileId)] = true;
+        }
+
+        let loadStaticFilesArgs = constructors.newGetLoadStaticFilesArgs(serviceId, staticFilesToCopyStringSet);
+        let request: grpc.requestCallback<bindingsJs.LoadStaticFilesResponse>;
+
+    	let loadStaticFilesResp, err = this.client.loadStaticFiles(loadStaticFilesArgs, request); //TODO - need to find another way to do error checking
+    	if (err != null) {
+    		return [null, new Error ("An error occurred loading the requested static files into the namespace of service '%v'")]; //TODO - remove the erorr, and need to be throwing back the erorr
+    	}
+    	let staticFileAbsFilepathsOnService = new Map(); //TODO (comment) = original type map[StaticFileID]string{}
+    	for (let staticFileId in loadStaticFilesResp.CopiedStaticFileRelativeFilepaths) { 
+    		let filepathRelativeToExVolRoot = loadStaticFilesResp.CopiedStaticFileRelativeFilepaths[staticFileId];
+            let absFilepathOnContainer = path.Join(this.testVolumeMountpointOnServiceContainer, filepathRelativeToExVolRoot)
+    		staticFileAbsFilepathsOnService[<StaticFileID>(staticFileId)] = absFilepathOnContainer; //TODO - see if I'm typecasting correctly
```

Comment:
Is this the right way to instantiate a new `StaticFileID`?

Reply:
Yep, I was a bit concerned about it as well, but I'm pretty sure this is one way to typecast using type assertions. Another way is using the `as` keyword. Here is the links I used: 
- https://www.typescriptlang.org/docs/handbook/basic-types.html#type-assertions
- https://www.tutorialsteacher.com/typescript/type-assertion